lib/stat_tracker.rb

```diff
@@ -136,11 +136,110 @@ def average_goals_by_season
     total_goals = total_goals_by_season
     count = count_of_games_by_season
     avg_season_goals = Hash.new(0.0)
-      total_goals.each do |season, goal|
-        avg_season_goals[season] = (goal / count[season]).round(2)
-      end
+    total_goals.each do |season, goal|
+      avg_season_goals[season] = (goal / count[season]).round(2)
+    end
     avg_season_goals
   end
 
-  #league stats
+  #season stats
+  def winningest_coach(season_id)
+    #first 4 char of season_id
+    games_id_year = season_id[0..3]
+
+    #hash with "coach" key and [0,0] value. first element is total games. second is games won
+    coach_stats = Hash.new { |coach, stats| coach[stats] = [0.0, 0.0] }
+    contents = CSV.open(@game_teams_path, headers: true, header_converters: :symbol)
+
+    contents.each do |row|
+      #iterates through every line checking to see if the game and season have the same 4 first chars
+      if row[:game_id][0..3] == games_id_year
+        coach_stats[row[:head_coach]][0] += 1
+        coach_stats[row[:head_coach]][1] += 1 if row[:result] == "WIN"
+      end
+    end
+
+    highest_win_percentage = 0.0
+    highest_win_percentage_coach = ""
+    #iterates through each coach and finding the highest win percentage
+    coach_stats.each do |coach, stats|
+      if highest_win_percentage < (stats[1] / stats[0])
+        highest_win_percentage = stats[1] / stats[0]
+        highest_win_percentage_coach = coach
+      end
+    end
+    highest_win_percentage_coach
+  end
+
+
+  def worst_coach(season_id)
+    #first 4 char of season_id
+    games_id_year = season_id[0..3]
+
+    #hash with "coach" key and [0,0] value. first element is total games. second is games won
+    coach_stats = Hash.new { |coach, stats| coach[stats] = [0.0, 0.0] }
+    contents = CSV.open(@game_teams_path, headers: true, header_converters: :symbol)
+
+    contents.each do |row|
+      #iterates through every line checking to see if the game and season have the same 4 first chars
+      if row[:game_id][0..3] == games_id_year
+        coach_stats[row[:head_coach]][0] += 1
+        coach_stats[row[:head_coach]][1] += 1 if row[:result] == "LOSS"
+      end
+    end
+
+    lowest_win_percentage = 1.0
+    lowest_win_percentage_coach = ""
+    #iterates through each coach and finding the lowest win percentage
+    coach_stats.each do |coach, stats|
+      if lowest_win_percentage > (stats[1] / stats[0])
+        lowest_win_percentage = stats[1] / stats[0]
+        lowest_win_percentage_coach = coach
+      end
+    end
+    lowest_win_percentage_coach
+  end
+
+  #*** returns hash {team_id => teamName}****
+  def team_names
```

Comment:
this could be really helpful in a few methods in the league statistics section to lesson the amount of lines used to achieve the same outcome!